stdlib/public/Concurrency/AsyncStreamBackpressure.swift

```diff
@@ -0,0 +1,1961 @@
+
+                return .callOnTermination(sourceFinished.onTermination)
+            }
+
+        case .finished:
+            // We are already finished so there is nothing left to clean up.
+            // This is just the references dropping afterwards.
+            return .none
+
+        case .modify:
+            fatalError("AsyncStream internal inconsistency")
+        }
+    }
+
+    mutating func iteratorInitialized() {
+        switch self._state {
+        case .initial(var initial):
+            if initial.iteratorInitialized {
+                // Our sequence is a unicast sequence and does not support multiple AsyncIterator's
+                fatalError("Only a single AsyncIterator can be created")
+            } else {
+                // The first and only iterator was initialized.
+                initial.iteratorInitialized = true
+                self._state = .initial(initial)
+            }
+
+        case .streaming(var streaming):
+            if streaming.iteratorInitialized {
+                // Our sequence is a unicast sequence and does not support multiple AsyncIterator's
+                fatalError("Only a single AsyncIterator can be created")
+            } else {
+                // The first and only iterator was initialized.
+                streaming.iteratorInitialized = true
+                self._state = .streaming(streaming)
+            }
+
+        case .sourceFinished(var sourceFinished):
+            if sourceFinished.iteratorInitialized {
+                // Our sequence is a unicast sequence and does not support multiple AsyncIterator's
+                fatalError("Only a single AsyncIterator can be created")
+            } else {
+                // The first and only iterator was initialized.
+                sourceFinished.iteratorInitialized = true
+                self._state = .sourceFinished(sourceFinished)
+            }
+
+        case .finished(iteratorInitialized: true):
+            // Our sequence is a unicast sequence and does not support multiple AsyncIterator's
+            fatalError("Only a single AsyncIterator can be created")
+
+        case .finished(iteratorInitialized: false):
+            // It is strange that an iterator is created after we are finished
+            // but it can definitely happen, e.g.
+            // Sequence.init -> source.finish -> sequence.makeAsyncIterator
+            self._state = .finished(iteratorInitialized: true)
+
+        case .modify:
+            fatalError("AsyncStream internal inconsistency")
+        }
+    }
+
+    /// Actions returned by `iteratorDeinitialized()`.
+    enum IteratorDeinitializedAction {
+        /// Indicates that `onTermination` should be called.
+        case callOnTermination((@Sendable () -> Void)?)
+        /// Indicates that  all producers should be failed and `onTermination` should be called.
+        case failProducersAndCallOnTermination(
+            [(Result<Void, Error>) -> Void],
```

Comment:
should this be an array? or should we just use the generic return from deque.map?

Reply:
The return from `Deque.map()` is an `Array` actually.